bitacross-worker/bitacross/core/bc-musig2-runner/src/lib.rs

```diff
@@ -1,429 +0,0 @@
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-//
-// Litentry is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
-//
-// You should have received a copy of the GNU General Public License
-// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
-
-#![cfg_attr(not(feature = "std"), no_std)]
-
-extern crate core;
-#[cfg(all(not(feature = "std"), feature = "sgx"))]
-extern crate sgx_tstd as std;
-
-#[cfg(all(feature = "std", feature = "sgx"))]
-compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
-
-use bc_enclave_registry::EnclaveRegistryLookup;
-use codec::Encode;
-use core::time::Duration;
-use itc_direct_rpc_client::{RpcClient, RpcClientFactory};
-use itc_direct_rpc_server::SendRpcResponse;
-use itp_ocall_api::EnclaveAttestationOCallApi;
-use itp_sgx_crypto::{
-	key_repository::{AccessKey, AccessPubkey},
-	ShieldingCryptoEncrypt,
-};
-use itp_utils::hex::ToHexPrefixed;
-use log::{debug, error, info, trace, warn};
-use sgx_crypto_helper::rsa3072::Rsa3072PubKey;
-use sp_core::{blake2_256, ed25519, Pair as SpCorePair, H256};
-use std::vec::Vec;
-
-#[cfg(feature = "std")]
-use std::sync::Mutex;
-
-use bc_musig2_ceremony::{
-	CeremonyCommandsRegistry, CeremonyEvent, CeremonyId, CeremonyRegistry, SignBitcoinError,
-	SignerId,
-};
-
-use itp_rpc::{Id, RpcRequest};
-use itp_sgx_crypto::schnorr::Pair as SchnorrPair;
-use itp_types::{DirectRequestStatus, Hash};
-use lc_direct_call::DirectCall;
-use litentry_primitives::{aes_encrypt_default, Address32, AesRequest, Identity, ShardIdentifier};
-#[cfg(feature = "sgx")]
-use std::sync::SgxMutex as Mutex;
-use std::{
-	collections::HashMap,
-	string::ToString,
-	sync::{mpsc::channel, Arc},
-	vec,
-};
-
-#[allow(clippy::too_many_arguments)]
-pub fn init_ceremonies_thread<ClientFactory, AK, ER, OCallApi, SIGNINGAK, SHIELDAK, Responder>(
-	signing_key_access: Arc<SIGNINGAK>,
-	shielding_key_access: Arc<SHIELDAK>,
-	client_factory: Arc<ClientFactory>,
-	enclave_registry: Arc<ER>,
-	ceremony_registry: Arc<Mutex<CeremonyRegistry<AK>>>,
-	ceremony_commands: Arc<Mutex<CeremonyCommandsRegistry>>,
-	ocall_api: Arc<OCallApi>,
-	responder: Arc<Responder>,
-) where
-	ClientFactory: RpcClientFactory + Send + Sync + 'static,
-	AK: AccessKey<KeyType = SchnorrPair> + Send + Sync + 'static,
-	ER: EnclaveRegistryLookup + Send + Sync + 'static,
-	OCallApi: EnclaveAttestationOCallApi + 'static,
-	SIGNINGAK: AccessKey<KeyType = ed25519::Pair> + Send + Sync + 'static,
-	SHIELDAK: AccessPubkey<KeyType = Rsa3072PubKey> + Send + Sync + 'static,
-	Responder: SendRpcResponse<Hash = H256> + 'static,
-{
-	let (responses_sender, responses_receiver) = channel();
-	std::thread::spawn(move || {
```

Comment:
What do you think about keeping this thread and processing ceremony commands sent from rpc handlers instead of ticking ceremonies ?

```
		while let Ok(command) = musig2_commands.recv() {
			// there is a command so get ceremony from registry and execute it, eventually process all events
		}
```

Reply:
I've already removed code about ticking, and this thread was kept actually, but merged into `bc-task-receiver` (now `bc-task-processor`), this module `bc-musig2-runner` (now `bc-musig2-event`) is not `runner` anymore.